dingo/gw/noise/synthetic_noise/asd_parameterization.py

```diff
@@ -0,0 +1,219 @@
+                    parameters_dict[det] = parameterize_asds_parallel(
+                        asds, domain, parameterization_settings, pool=pool, verbose=verbose
+                    )
+        else:
+            parameters_dict[det] = parameterize_asds_parallel(asds, domain, parameterization_settings)
+
+    return parameters_dict
+
+
+def parameterize_asds_parallel(asds, domain, parameterization_settings, pool=None, verbose=False):
+
+    task_func = partial(parameterize_single_psd, domain=domain, parameterization_settings=parameterization_settings)
+    psds = asds ** 2
+    if pool is not None:
+        parameters_list = list(tqdm.tqdm(pool.imap(task_func, psds), total=psds.shape[0], disable=not verbose))
+    else:
+        parameters_list = list(tqdm.tqdm(map(task_func, psds), total=psds.shape[0], disable=not verbose))
+
+    parameters = {
+        feature: np.stack([asd_param[feature] for asd_param in parameters_list])
+        for feature in parameters_list[0].keys()
+    }
+    # the x-positions are the same for all asds
+    parameters["x_positions"] = parameters["x_positions"][0]
+
+    return parameters
+
+
+def parameterize_single_psd(real_psd, domain, parameterization_settings):
+    sigma = float(parameterization_settings["sigma"])
+    if not len(real_psd) == len(domain.sample_frequencies):
+        real_psd = domain.update_data(real_psd)
+
+    # optional truncation parameter for Lorentzians. Set to None if non-positive value is passed
+    delta_f = parameterization_settings.get("delta_f", -1)
+    delta_f = float(delta_f) if delta_f > 0 else None
+
+    # transform psd to log space
+    transformed_psd = np.log(real_psd)
+
+    # parameterize broad-band noise
+    xs, ys = fit_broadband_noise(
+        domain=domain,
+        psd=transformed_psd,
+        num_spline_positions=parameterization_settings["num_spline_positions"],
+        sigma=sigma,
+    )
+    spline = scipy.interpolate.CubicSpline(xs, ys)
+    broadband_noise = spline(domain.sample_frequencies)
+
+    _, features = fit_spectral(
+        frequencies=domain.sample_frequencies,
+        psd=transformed_psd,
+        broadband_noise=broadband_noise,
+        num_spectral_segments=parameterization_settings["num_spectral_segments"],
+        sigma=sigma,
+        delta_f=delta_f,
+    )
+    # features = np.random.normal(100, 10, size=(400, 3))
+    parameter_dict = {"x_positions": xs, "y_values": ys, "spectral_features": features}
+
+    return parameter_dict
+
+
+def fit_broadband_noise(domain, psd, num_spline_positions, sigma):
+    frequencies = domain.sample_frequencies
+
+    # standardize frequencies to the interval [0,1]
+    standardized_frequencies = (frequencies - frequencies[0]) / (
+        frequencies[-1] - frequencies[0]
+    )
+
+    # log-distributed x-positions in [20/f_max, 1] for the interpolating base noise spline
+    log_xs = np.logspace(np.log10(F_MIN / domain.f_max), 0, num_spline_positions)
```

Comment:
What is the reason to hard-code `F_MIN=20`? What if we want to analyze data going below 20 Hz?

Reply:
This needs to be >0 due to the log distribution, and usually the ASD is very unstable in the low frequency regions. For convenience, we just picked 20 at some point for our applications. The ASDs will still include values below 20, but these won't be fitted very accurately. I agree that this could be generalised at some point

Reply:
Okay, we should note to extend this, since at some point we may want to go below 20 Hz. Could you create an Issue so we can remember?

Reply:
https://github.com/dingo-gw/dingo/issues/182